File: BackEnd/BusinessLogic/Models/ChatInfo.cs
namespace ThirdLayer.Models
{
    public class ChatInfo
    {
        public ChatInfo() { }
        public int ChatId { get; set; }
        public int ChatUserId { get; set; }
        public String UserFirstName { get; set; }
        public String UserLastName { get; set; }
        public String UserImage { get; set; }
        public bool UserGender { get; set; }
        public String UserLastMessage { get; set; }
        public DateTime LastMessageDate { get; set; }
        public int NumberOfUnreadMessage { get; set; }
    }
}

File: UI/doctor/lib/data/models/chatInfo.dart
class ChatInfo {
  int? chatId;
  int chatUserId;
  String userFirstName;
  String userLastName;
  String? userImage;
  bool userGender;
  String? userLastMessage;
  DateTime? lastMessageDate;
  int? numberOfUnreadMessage;

  ChatInfo(
      { this.chatId,
      required this.chatUserId,
      required this.userFirstName,
      required this.userLastName,
      this.userImage,
      required this.userGender,
      this.userLastMessage,
      this.lastMessageDate,
      this.numberOfUnreadMessage});

  factory ChatInfo.fromJson(Map<String, dynamic> json) => ChatInfo(
      chatId: json["chatId"],
      userFirstName: json["userFirstName"],
      userLastName: json["userLastName"],
      userImage: json["userImage"],
      userGender: json["userGender"],
      userLastMessage: json["userLastMessage"],
      lastMessageDate: DateTime.parse(json["lastMessageDate"]),
      numberOfUnreadMessage: json["numberOfUnreadMessage"],
      chatUserId: json["chatUserId"]);

  Map<String, dynamic> toJson() => {
        "chatId": chatId,
        "userFirstName": userFirstName,
        "userLastName": userLastName,
        "userImage": userImage,
        "userGender": userGender,
        "userLastMessage": userLastMessage,
        "lastMessageDate": lastMessageDate?.toIso8601String(),
        "numberOfUnreadMessage": numberOfUnreadMessage,
        "chatUserId": chatUserId
      };
}
